### PR TITLE
Add a feature flag to enable using URLSession in WordPressKit

### PR DESCRIPTION
**Pull request by crazytonyli**
The feature flag is enabled for all non-production builds.

## Test Instructions

With the feature flag turned on, try features that involves different kinds of HTTP requests, i.e. using self hosted sites, uploading an image/video, deleting a post, updating a post, etc.

Testing error cases would be super valuable too.

## Regression Notes
1. Potential unintended areas of impact
None.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
What's described in the Test Instructions section.

3. What automated tests I added (or what prevented me from doing so)
None.

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

UI Changes testing checklist: N/A

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">PR requires at least one label.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WordPress/Classes/Utility/BuildInformation/FeatureFlag.swift [hunk 1]
```diff
@@ -37,6 +38,8 @@ enum FeatureFlag: Int, CaseIterable {
             return false
         case .newTabIcons:
             return true
+        case .useURLSession:
+            return BuildConfiguration.current != .appStore
```
Comment (mokagio) [inline WordPress/Classes/Utility/BuildInformation/FeatureFlag.swift:42]: Did you consider making this a remote flag?

File: WordPress/Classes/System/WordPressAppDelegate.swift [hunk 2]
```diff
@@ -133,6 +133,11 @@ class WordPressAppDelegate: UIResponder, UIApplicationDelegate {
 
         ABTest.start()
 
+        if FeatureFlag.useURLSession.enabled {
+            WordPressComRestApi.useURLSession = true
+            WordPressOrgXMLRPCApi.useURLSession = true
+        }
```
Comment (mokagio) [inline WordPress/Classes/System/WordPressAppDelegate.swift]: Nitpick: I liked how the WooCommerce counterpart has this in a dedicated function. I feel like it's a great way to keep the configuration cod eat this level tidier.

**Review by mokagio — APPROVED** [review]

File: WordPress/Classes/Utility/BuildInformation/FeatureFlag.swift (hunk 1 above)
Comment (crazytonyli) [reply WordPress/Classes/Utility/BuildInformation/FeatureFlag.swift:42]: See https://github.com/woocommerce/woocommerce-ios/pull/11927#discussion_r1482119827

File: WordPress/Classes/System/WordPressAppDelegate.swift (hunk 2 above)
Comment (crazytonyli) [reply WordPress/Classes/System/WordPressAppDelegate.swift]: 👍 Addressed in https://github.com/wordpress-mobile/WordPress-iOS/pull/22540/commits/1142472c0cb2b2a1b5acbba0e51a2512792aec8f

**Pull request merged**